modules/git/repo_branch.go

```diff
@@ -179,6 +179,22 @@ func (repo *Repository) RemoveRemote(name string) error {
 	return err
 }
 
+// HasRemote checks if a repository has a specified remote.
+func (repo *Repository) HasRemote(name string) (bool, error) {
```

Comment:
What is this function for?

Reply:
originaly it was for the old way to fix `GetDiverging` now it's only used in the tests to validate that the other code is working. could be used in future ....

Reply:
I remember a PR of yours not very long ago removing unused code... 😁

Reply:
This function does not seem used anymore

Reply:
ok I'll remove it ... tests will be gone too then ...